react_components/PaginationBoxView.js

```diff
@@ -38,30 +38,50 @@ var PaginationBoxView = React.createClass({
     };
   },
 
-  handlePageSelected: function(selected, event) {
-    event.preventDefault();
-
-    if (this.state.selected === selected) return;
+  callClickCallback: function() {
+    if (typeof this.props.clickCallback !== "undefined" &&
+        typeof this.props.clickCallback === "function") {
+      this.props.clickCallback({ selected: this.state.selected });
+    }
+  },
 
-    this.setState({selected: selected});
+  handlePageSelected: function(selected, event) {
+    if (typeof event !== "undefined") {
+      event.preventDefault();
+    }
 
-    if (typeof(this.props.clickCallback) !== "undefined" &&
-        typeof(this.props.clickCallback) === "function") {
-      this.props.clickCallback({selected: selected});
+    var needToUpdate = this.state.selected !== selected;
+    if (needToUpdate) {
+      this.setState({ selected: selected }, this.callClickCallback);
+    }
+    else {
+      this.callClickCallback();
```

Comment:
I understand why you want to use setState's callback here but I think we don't need to call the 'callClickCallback' function if the variable 'selected' equals the previous state.selected.